scalding-spark/src/main/scala/com/twitter/scalding/spark_backend/SparkMode.scala

```diff
@@ -3,18 +3,23 @@ package com.twitter.scalding.spark_backend
 import com.twitter.scalding.{ Config, Mode }
 import com.twitter.scalding.typed.{ Resolver, TypedSource, TypedSink }
 import org.apache.spark.rdd.RDD
-import org.apache.spark.SparkContext
+import org.apache.spark.{ SparkContext, SparkConf }
```

Comment:
if your defining these things, these days i'd take a SparkSession than a spark context, you can get a context from a session but not vice versa

Reply:
wouldn't that suggest we should take the least power, not most power? For instance, if you have a session, you can make a SparkMode now, but if you have a context you couldn't if I require session.

Reply:
Well except it would mean users need to thread both through, a bunch of API's say for doing I/O will want a session. So using any of the datasource v2 api's or to write out via a dataset. I think the view is normally that SparkSession is the one you hand about.


(Though there is a default global static one you can get access, so it is workaround able to grab that if a user doesn't have multiple sessions in flight)

Reply:
okay, can I make this change in a follow up PR? I didn't know that it was better to require SparkSession.

Also, I guess it is better since you can use sql first to produce smarter sources, which I think needs the Session.

Can I merge this and follow up?

Reply:
Yep on all fronts -- I think you'll get into the DatasourceV2 stuff to do custom stuff before long.

but totally do in a follow up, its all alpha anyway! :)